### PR TITLE
transition yaml codemods to new api

### PR DESCRIPTION
**Pull request by clavedeluna**
## Overview
*This PR moves both pyyaml and ruamel codemods to use the new api*

## Description

* API change: `update_arg_target` takes a list of new args

## Describe your testing approach

* Happy/sad paths, Edge cases, Bad inputs
* Explanation of manual testing with any TS screenshots
* Or, an explanation of why this change does not need tests

## Performance

* How do you expect this PR to affect performance?

## Additional Details
* any follow up tickets or discussion
* Any specific merge / deploy details

### CODE REVIEW

File: codemodder/codemods/harden_pyyaml.py [hunk 1]
```diff
@@ -1,53 +1,45 @@
...
+class HardenPyyaml(SemgrepCodemod):
+    NAME = "harden-pyyaml"
+    REVIEW_GUIDANCE = ReviewGuidance.MERGE_WITHOUT_REVIEW
+    DESCRIPTION = "Ensures all calls to yaml.load use `SafeLoader`."
+
+    @classmethod
+    def rule(cls):
+        return """
+            rules:
+              - id: harden-pyyaml
```
Comment (drdavella) [inline codemodder/codemods/harden_pyyaml.py]: Can we omit the `id` field here?

File: codemodder/codemods/harden_pyyaml.py [hunk 2]
```diff
@@ -1,53 +1,45 @@
...
+                      - metavariable-pattern:
+                          metavariable: $ARG
+                          patterns:
+                            - pattern-not:
+                                pattern: yaml.SafeLoader
+
+        """
+
+    def on_result_found(self, _, updated_node):
+        second_arg = cst.parse_expression("yaml.SafeLoader")
```
Comment (drdavella) [inline codemodder/codemods/harden_pyyaml.py]: Just a small thing but I think we should hide this behind a helper so that the user doesn't have to make reference to `libCST` if it's not necessary. It could maybe just become `self.parse_expression`.

File: codemodder/codemods/harden_ruamel.py [hunk 3]
```diff
@@ -1,52 +1,38 @@
...
-        return updated_node
+        """
+
+    def on_result_found(self, original_node, updated_node):
+        new_args = make_new_args(original_node.args, target_arg="typ")
+        return self.update_arg_target(updated_node, new_args)
 
 
-def update_arg_target(original_args, target_arg):
+def make_new_args(original_args, target_arg):
```
Comment (drdavella) [inline codemodder/codemods/harden_ruamel.py]: Is this more generally useful now? Could it move to the `Helpers` class?

File: codemodder/codemods/limit_readline.py [hunk 4]
```diff
@@ -35,4 +35,4 @@ def rule(cls):
         """
 
     def on_result_found(self, _, updated_node):
-        return self.update_arg_target(updated_node, cst.Integer(default_limit))
+        return self.update_arg_target(updated_node, [cst.Integer(default_limit)])
```
Comment (drdavella) [inline codemodder/codemods/limit_readline.py:38]: It probably wouldn't be too hard to keep the old API for single arguments but I don't feel too strongly about it right now.

File: codemodder/codemods/harden_pyyaml.py (hunk 1 above)
Comment (drdavella) [reply codemodder/codemods/harden_pyyaml.py]: Also I think I was guilty of this myself in my original PR, but maybe we should remove the YAML files now.

**Review by drdavella — CHANGES_REQUESTED** [review]
Looks good overall but just a handful of small comments.

File: codemodder/codemods/limit_readline.py (hunk 4 above)
Comment (clavedeluna) [reply codemodder/codemods/limit_readline.py:38]: I thought about that too but it felt more general to say we can support a list of args instead of one arg only.

File: codemodder/codemods/harden_pyyaml.py (hunk 2 above)
Comment (clavedeluna) [reply codemodder/codemods/harden_pyyaml.py]: I like this thinking. Basically if a codemod has to `import cst` then we can likely move it to a helper.

File: codemodder/codemods/harden_ruamel.py (hunk 3 above)
Comment (clavedeluna) [reply codemodder/codemods/harden_ruamel.py]: good suggestion, did that

**Review by andrecsilva — APPROVED** [review]

**Review by drdavella — APPROVED** [review]
